### ISSUE TITLE
GPU inference skips per-head Q/K normalisation (Qwen3-style models diverge from CPU)

### ISSUE THREAD

**Issue by MBrekhof** [Integral2u/SharpMind#39]
`GpuInferenceEngine` goes straight from the Q/K projections to RoPE. `GpuBlock` never uploads `QNormW` / `KNormW`, and `ValidateSupported` does not reject models that carry them (it does reject the Gemma post-attention/post-FFN norms, so the gate exists, just not for these two). The CPU `AttentionLayer` applies the per-head RMS norms to Q and K before RoPE.

Affected: every model with per-head Q/K norms — the Qwen3 family in particular, which the CPU path supports. The omission is in all three GPU forward paths (first prefill, continued prefill, decode).

**Repro** (synthetic):

```csharp
var cfg = new ModelConfig { VocabSize = 256, HiddenDim = 64, NumLayers = 2, NumHeads = 4, NumKvHeads = 2, FfnDim = 128, MaxSeqLen = 64, Architecture = "qwen2" };
var config = SharpMindConfig.Llama with { Hardware = HardwareTier.Scalar };
var weights = ModelFactory.CreateForTraining(cfg, config);
WeightInitializer.InitializeRandomly(weights, 9001);
foreach (var block in weights.Blocks)
{
    block.QNormW = Tensor<float>.Ones(cfg.HeadDim);
    block.KNormW = Tensor<float>.Ones(cfg.HeadDim);
}

using var model = ModelFactory.CreateTrainingTransformer(weights, config);
using var device = GpuDevice.Create();
using var engine = new GpuInferenceEngine(device, model, config, 64, 32);
int[] ids = [65, 66, 67, 32, 68, 69];

// CPU logits change vs. the no-norm control (max |logit| 0.741 vs 0.827)
// GPU logits are bit-identical to the no-norm control
// max |CPU - GPU| = 0.186
```

Unit norm weights still change the result because the RMS normalisation itself rescales each head, so "identical to the unnormalised control" is the signature of the norm being skipped rather than a numerical difference.

**Suggested fix:** upload `QNormW`/`KNormW` per block and apply a per-head RMS norm kernel to Q and K after the projections and before `RopeFwd`, in all three paths; or, until then, have `ValidateSupported` throw on `block.QNormW is not null || block.KNormW is not null` so these models stay on CPU.

Observed on OpenCL (gfx1152); backend-independent.

**Comment by Integral2u** [comment]
Unfortunately, I cannot seem to get QNormW/KNormW working on this system, quite possibly a limitation of the system.  I will have to go the rout of throw on block.QNormW is not null || block.KNormW is not null.

**Comment by Integral2u** [comment]
Move to GPU inference per-head Q/K normalization #44
see commit 41c6edcb 10/9/26
